spec/spec_helper.rb

```diff
@@ -58,6 +58,7 @@
   config.before :each do
     WebMock.disable_net_connect!
     allow(JobBase).to receive(:enqueue_in)
+    allow_any_instance_of(Build).to receive(:ensure_initiated_by).and_return(nil)
```

Comment:
there is another callback on build that is being run now too `enqueue_partitioning_job`. That job is just putting some data in redis which gets flushed almost immediately so it does not have a side effect for the tests. However, it may speed up the tests a little bit to stub it out.